__tests__/hooks/headless/useChatMessages.test.tsx

```diff
@@ -0,0 +1,162 @@
+import { MedplumClient } from "@medplum/core";
+import { Communication, Patient } from "@medplum/fhirtypes";
+import { MockClient } from "@medplum/mock";
+import { MedplumProvider } from "@medplum/react-hooks";
+import { act, renderHook, waitFor } from "@testing-library/react-native";
+
+import { useChatMessages } from "@/hooks/headless/useChatMessages";
+
+const mockPatient: Patient = {
+  resourceType: "Patient",
+  id: "test-patient",
+  name: [{ given: ["John"], family: "Doe" }],
+};
+
+const mockThread: Communication = {
+  resourceType: "Communication",
+  id: "test-thread",
+  status: "completed",
+  sent: "2024-01-01T12:00:00Z",
+};
+
+const mockMessage1: Communication = {
+  resourceType: "Communication",
+  id: "msg-1",
+  status: "completed",
+  sent: "2024-01-01T12:01:00Z",
+  sender: {
+    reference: "Patient/test-patient",
+    display: "John Doe",
+  },
+  payload: [{ contentString: "Hello" }],
+  partOf: [{ reference: "Communication/test-thread" }],
+};
+
+const mockMessage2: Communication = {
+  resourceType: "Communication",
+  id: "msg-2",
+  status: "completed",
+  sent: "2024-01-01T12:02:00Z",
+  sender: {
+    reference: "Practitioner/test-doctor",
+    display: "Dr. Smith",
+  },
+  payload: [{ contentString: "Hi there" }],
+  partOf: [{ reference: "Communication/test-thread" }],
+};
+
+describe("useChatMessages", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  async function setup(): Promise<MedplumClient> {
+    const medplum = new MockClient({ profile: mockPatient });
+
+    // Setup mock data
+    await medplum.createResource(mockPatient);
+    await medplum.createResource(mockThread);
+    await medplum.createResource(mockMessage1);
+    await medplum.createResource(mockMessage2);
+
+    return medplum as unknown as MedplumClient;
+  }
```

Comment:
Hmm, did you get any errors when using `MockClient` here instead of `MedplumClient`?

Reply:
It's probably a config thing. I experienced a similar problem with `vitest`, and adding `"types": ["vitest/globals"]` to `tsconfig.json` resolved it, but I am not sure which is the fix for `jest`.

Do you have the Jest extension installed on your IDE?

Reply:
it was a version problem, now it's fixed (I had the differing versions of medplum libs)